vllm/transformers_utils/tokenizer.py

```diff
@@ -152,6 +153,27 @@ def get_tokenizer(
             else:
                 raise e
 
+        if type(tokenizer).__name__ == "ChatGLMTokenizer":
+            assert isinstance(tokenizer, PreTrainedTokenizer)
+            orig_pad = tokenizer._pad
+
+            # Patch _pad method to accept `padding_side`
+            def _pad(
+                self: PreTrainedTokenizer,
+                *args,
+                padding_side: Optional[str] = None,
+                **kwargs,
+            ):
+                if (padding_side is not None
+                        and padding_side != self.padding_side):
+                    msg = ("`padding_side` argument is not supported by "
+                           "ChatGLMTokenizer and will be ignored.")
+                    warnings.warn(msg, stacklevel=2)
+
+                return orig_pad(*args, **kwargs)
+
+            tokenizer._pad = MethodType(_pad, tokenizer)
+
```

Comment:
We also need to add patch for glm4's tokenizer, see： https://huggingface.co/THUDM/glm-4-9b-chat/blob/main/tokenization_chatglm.py#L261

Reply:
Can you link me to the relevant issues on HF so I can reference them in the code?

Reply:
Nvm, found it